stackit/internal/services/iaas/network/resource.go

```diff
@@ -248,6 +249,21 @@ func (r *networkResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 	}
 }
 
+// ModifyPlan will be called in the Plan phase and will check if the plan is a creation of the resource
+// and if nameserver and ipv4_nameserver have been provided at the same time.
+// If so, it will show an error
+func (r *networkResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) { // nolint:gocritic // function signature required by Terraform
+	if req.State.Raw.IsNull() {
+		// Planned to create a network resource
+		var plan Model
+		_ = req.Plan.Get(ctx, &plan)
+
+		if !plan.Nameservers.IsUnknown() && !plan.IPv4Nameservers.IsUnknown() {
+			core.LogAndAddError(ctx, &resp.Diagnostics, "Planned to create network resource", "You have provided nameservers and ipv4_nameservers field at the same time. Nameservers field has been deprecated. Please remove this field and try again.")
+		}
+	}
+}
```

Comment:
Why do we need the `if req.State.Raw.IsNull()` here? We could just check the plan and see it both `Nameservers` and `IPv4Nameservers` were provided, and if yes throw the error

Reply:
Then we wouldn't need the check in the `Update` function

Reply:
I tried this, but in that case, it throws error when we trigger delete operation also

Reply:
I have changed the conditions to handle, create // update // delete at the same time. It works for all. 
Only if I have removed nameservers and ipv4_nameserver fields at the same time, it gives alert that I provided both fields. Needs to fix that